test/connection/PipelinedWriterTest.js

```diff
@@ -25,22 +25,35 @@ const { DeferredPromise } = require('../../lib/util/Util');
 const { PipelinedWriter } = require('../../lib/network/ClientConnection');
 
 describe('PipelinedWriterTest', function () {
+
     let writer;
     let mockSocket;
 
-    const setUpWriteSuccess = () => {
+    function setUpWriteSuccess() {
```

Comment:
Minor: I think we can get rid of the duplication from `setUpWriteSuccess` and `setUpWriteSuccessWaitForDrain` by making one of them take a boolean parameter and return it. Something like
```node
    function setUpWrite(shouldWrite) {
        mockSocket = new Socket({});
        sinon.stub(mockSocket, 'write').callsFake((data, cb) => {
            process.nextTick(cb);
            process.nextTick(() => mockSocket.emit('data', data));
            return shouldWrite;
        });
        writer = new PipelinedWriter(mockSocket, 8192);
    }
```
What do you think about it?

Reply:
Good suggestion. Merged the functions in 3aa1ab04a696800623afd5f7d88459c55649097e